sdp/processors/datasets/fleurs/create_initial_manifest.py

```diff
@@ -0,0 +1,160 @@
+
+    def process_transcript(self, file_path: str) -> list[dict[str, typing.Any]]:
+        """
+        Parse transcript TSV file and put it inside manyfest.
+        Assumes the TSV file has two columns: file name and text.
+        """
+
+        entries = []
+        root = os.path.dirname(file_path)
+
+        with open(file_path, encoding="utf-8") as fin:
+            for line in fin:
+                # Split the line into filename text using the tab delimiter
+                parts = line.strip().split('\t')
+                if len(parts) < 2:  # Skip lines that don't have at least 2 parts
+                    continue
+
+                file_name, transcript_text = parts[1], parts[2]
+                wav_file = os.path.join(root, file_name)
+
+                entry = {"audio_filepath": os.path.abspath(wav_file), "text": transcript_text}
+                entries.append(entry)
+
+        return entries
+
+    def process_data(self, data_folder: str, manifest_file: str) -> None:
+        files = []
+        entries = []
+
+        for root, _, filenames in os.walk(data_folder):
+            for filename in fnmatch.filter(filenames, "*.tsv"):
+                files.append(os.path.join(root, filename))
+
+        for file in files:
+            result = self.process_transcript(file)
+            entries.extend(result)
+
+        with open(manifest_file, "w", encoding="utf-8") as fout:
+            for m in entries:
+                fout.write(json.dumps(m, ensure_ascii=False) + "\n")
+
+    def download_extract_files(self, dst_folder: str) -> None:
+        """downloading and extracting files"""
+
+        os.makedirs(dst_folder, exist_ok=True)
+
+        # downloading all files
+        for file_url in get_fleurs_url_list(self.lang, self.split):
+            download_file(file_url, str(dst_folder))
+        for data_file in glob.glob(f'{dst_folder}/*.tar.gz'):
+            extract_archive(str(data_file), str(dst_folder), force_extract=True)
+
+        # Organizing files into their respective folders
+        folder_mappings = {'test': '*test.tsv', 'train': '*train.tsv', 'dev': '*dev.tsv'}
+
+        for folder, file_pattern in folder_mappings.items():
+            target_folder = os.path.join(dst_folder, folder)
+
+            if not os.path.exists(target_folder):
+                continue
+
+            all_files = os.listdir(dst_folder)
+
+            matching_files = fnmatch.filter(all_files, file_pattern)
+
+            for file_name in matching_files:
+                file_path = os.path.join(dst_folder, file_name)
+                dest_file_path = os.path.join(target_folder, file_name)
+                if not os.path.exists(dest_file_path):
+                    shutil.move(file_path, dest_file_path)
+                    print(f'Moved {file_path} to {dest_file_path}')
+                else:
+                    os.remove(file_path)
+                    print(f'File {file_name} already exists in {target_folder}, deleted from source.')
```

Comment:
Not sure that we really need this?

Reply:
This is for testing purposes mainly. Since for test we use custom manifest file(that contains few lines) instead of big manifests like dev, test, train, we need to make sure during testing our custom manifest does not get replaced and we don't get a new manifest file without any purpose.